From bba5f85bf24ca0847fcf3f4e1c21565168ea555e Mon Sep 17 00:00:00 2001
From: Jarek Jarcec Cecho <jarcec@apache.org>
Date: Wed, 1 May 2013 08:03:02 -0700
Subject: [PATCH 104/121] CLOUDERA-BUILD. Need capability to specify the target bytecode to the compiler JDK1.7.

CDH-11658: Hive - Need capability to specify the target bytecode to the compiler JDK1.7.
---
 ant/build.xml       |    3 +++
 beeline/build.xml   |    3 +++
 build.xml           |   11 +++++++++++
 cli/build.xml       |    3 +++
 common/build.xml    |    3 +++
 hwi/build.xml       |    3 +++
 jdbc/build.xml      |    4 ++++
 metastore/build.xml |    6 ++++++
 pdk/build.xml       |    3 +++
 ql/build.xml        |    3 +++
 serde/build.xml     |    3 +++
 service/build.xml   |    3 +++
 shims/build.xml     |    3 +++
 13 files changed, 51 insertions(+), 0 deletions(-)

diff --git a/src/ant/build.xml b/src/ant/build.xml
index e23d6dd..f54634b 100644
--- a/src/ant/build.xml
+++ b/src/ant/build.xml
@@ -35,6 +35,9 @@ to call at top-level: ant deploy-contrib compile-core-test
      includes="**/*.java"
      destdir="${build.classes}"
      debug="${javac.debug}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      deprecation="${javac.deprecation}">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/beeline/build.xml b/src/beeline/build.xml
index d583915..e72147d 100644
--- a/src/beeline/build.xml
+++ b/src/beeline/build.xml
@@ -36,6 +36,9 @@ to call at top-level: ant deploy-contrib compile-core-test
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/build.xml b/src/build.xml
index 1ed8787..2a9fdc9 100644
--- a/src/build.xml
+++ b/src/build.xml
@@ -77,6 +77,9 @@
   <property name="images.src" value="${docs.src}/images"/>
   <property name="javadoc.link.java"
             value="http://java.sun.com/javase/6/docs/api/"/>
+  <property name="sourceJavaVersion" value="${javac.version}" />
+  <property name="targetJavaVersion" value="${javac.version}" />
+  <property name="javaBootClasspath" value="${java.home}/jre/lib/rt.jar" />
   <property name="final.name" value="${name}-${version}"/>
   <property name="tar.final.name" value="${final.name}"/>
   <property name="bin.final.name" value="${final.name}-bin"/>
@@ -181,6 +184,9 @@
       <subant target="@{target}">
         <property name="build.dir.hive" location="${build.dir.hive}"/>
         <property name="is-offline" value="${is-offline}"/>
+        <property name="sourceJavaVersion" value="${sourceJavaVersion}" />
+        <property name="targetJavaVersion" value="${targetJavaVersion}" />
+        <property name="javaBootClasspath" value="${javaBootClasspath}" />
         <filelist dir="." files="shims/build.xml,common/build.xml,serde/build.xml,metastore/build.xml,ql/build.xml,contrib/build.xml,service/build.xml,cli/build.xml,jdbc/build.xml,beeline/build.xml,hwi/build.xml,hbase-handler/build.xml,pdk/build.xml,builtins/build.xml,testutils/build.xml"/>
       </subant>
     </sequential>
@@ -302,6 +308,8 @@
     <echo message="Project: ${ant.project.name}"/>
     <subant target="compile">
       <fileset dir="." includes="ant/build.xml"/>
+      <property name="sourceJavaVersion" value="${sourceJavaVersion}"/>
+      <property name="targetJavaVersion" value="${targetJavaVersion}"/>
     </subant>
   </target>
 
@@ -372,6 +380,9 @@
         <ant antfile="@{file}" target="test" inheritAll="false" inheritRefs="true">
           <property name="build.dir.hive" location="${build.dir.hive}"/>
           <property name="is-offline" value="${is-offline}"/>
+          <property name="sourceJavaVersion" value="${sourceJavaVersion}" />
+          <property name="targetJavaVersion" value="${targetJavaVersion}" />
+          <property name="javaBootClasspath" value="${javaBootClasspath}" />
         </ant>
       </sequential>
     </for>
diff --git a/src/cli/build.xml b/src/cli/build.xml
index 092a68b..32504d1 100755
--- a/src/cli/build.xml
+++ b/src/cli/build.xml
@@ -36,6 +36,9 @@ to call at top-level: ant deploy-contrib compile-core-test
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/common/build.xml b/src/common/build.xml
index 731f26e..13dba33 100755
--- a/src/common/build.xml
+++ b/src/common/build.xml
@@ -41,6 +41,9 @@ to call at top-level: ant deploy-contrib compile-core-test
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/hwi/build.xml b/src/hwi/build.xml
index e982ebe..4921d64 100644
--- a/src/hwi/build.xml
+++ b/src/hwi/build.xml
@@ -64,6 +64,9 @@
      includes="**/*.java"
      destdir="${build.classes}"
      debug="${javac.debug}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      deprecation="${javac.deprecation}">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/jdbc/build.xml b/src/jdbc/build.xml
index b75c38f..b314bfd 100644
--- a/src/jdbc/build.xml
+++ b/src/jdbc/build.xml
@@ -43,6 +43,7 @@
 
   <target name="compile" depends="init,ivy-retrieve">
     <echo message="Project: ${ant.project.name}"/>
+    <echo message="JDK source: ${sourceJavaVersion}, target: ${targetJavaVersion}, bootclasspath: ${javaBootClasspath}"/>
     <javac
         encoding="${build.encoding}"
         srcdir="${src.dir}/java"
@@ -50,6 +51,9 @@
         destdir="${build.classes}"
         debug="${javac.debug}"
         deprecation="${javac.deprecation}"
+        bootclasspath="${javaBootClasspath}"
+        source="${sourceJavaVersion}"
+        target="${targetJavaVersion}"
         includeantruntime="false"
         >
       <classpath refid="classpath"/>
diff --git a/src/metastore/build.xml b/src/metastore/build.xml
index 4779d70..652d87f 100755
--- a/src/metastore/build.xml
+++ b/src/metastore/build.xml
@@ -51,6 +51,9 @@
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <classpath refid="classpath"/>
     </javac>
@@ -66,6 +69,9 @@
         srcdir="${model.dir}"
         destdir="${build.classes}"
         debug="${javac.debug}"
+        source="${sourceJavaVersion}"
+        target="${targetJavaVersion}"
+        bootclasspath="${javaBootClasspath}"
         includeantruntime="false">
        <classpath refid="classpath"/>
     </javac>
diff --git a/src/pdk/build.xml b/src/pdk/build.xml
index 0893f47..62b14f0 100644
--- a/src/pdk/build.xml
+++ b/src/pdk/build.xml
@@ -35,6 +35,9 @@
      includes="**/*.java"
      destdir="${build.classes}"
      debug="${javac.debug}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      deprecation="${javac.deprecation}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
diff --git a/src/ql/build.xml b/src/ql/build.xml
index ba6e288..70f232e 100644
--- a/src/ql/build.xml
+++ b/src/ql/build.xml
@@ -179,6 +179,9 @@
      includes="**/*.java"
      destdir="${build.classes}"
      debug="${javac.debug}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      deprecation="${javac.deprecation}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
diff --git a/src/serde/build.xml b/src/serde/build.xml
index a2c23d1..9cd3f00 100644
--- a/src/serde/build.xml
+++ b/src/serde/build.xml
@@ -45,6 +45,9 @@
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="classpath"/>
diff --git a/src/service/build.xml b/src/service/build.xml
index cb0beb4..f138dd7 100644
--- a/src/service/build.xml
+++ b/src/service/build.xml
@@ -31,6 +31,9 @@
      destdir="${build.classes}"
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <classpath refid="classpath"/>
     </javac>
diff --git a/src/shims/build.xml b/src/shims/build.xml
index 529e846..0aaa8be 100644
--- a/src/shims/build.xml
+++ b/src/shims/build.xml
@@ -62,6 +62,9 @@ to call at top-level: ant deploy-contrib compile-core-test
      debug="${javac.debug}"
      deprecation="${javac.deprecation}"
      srcdir="${sources}"
+     source="${sourceJavaVersion}"
+     target="${targetJavaVersion}"
+     bootclasspath="${javaBootClasspath}"
      includeantruntime="false">
       <compilerarg line="${javac.args} ${javac.args.warnings}" />
       <classpath refid="shims.classpath"/>
-- 
1.7.0.4

